app/assets/javascripts/components/houses-map.js

```diff
@@ -1,39 +1,124 @@
 class HousesMap {
   constructor(params={}) {
     this.mapParent = params.map || document.getElementById('map')
+    this.houses = params.houses
+    this.housesFilters = params.housesFilters
     this.markers = []
+    this.map
+    this.placesService
   }
 
-  initMap() {
+  async initMap(coordinates) {
+    this._removeMap()
+    let initialCoords = coordinates || this._getInitialCoordinates()
     this.map = new google.maps.Map(this.mapParent, {
-      center: new google.maps.LatLng(28.39404819, -91.38743867),
-      zoom: 8,
+      center: initialCoords,
+      zoom: 12,
     })
+    this.placesService = new google.maps.places.PlacesService(this.map)
   }
 
-  render(query) {
-    this._resetMapChildren()
-    this.initMap() 
-    if (query) { this.placeMarkers(query) }
+  query(input) {
+    let req = { query: input, fields: ['geometry'] }
+    this.placesService.findPlaceFromQuery(req, this.findAndRenderHouses.bind(this))
   }
 
-  async placeMarkers(query) {
-    let houses = await HousesService.search(query)  
-    if (!houses) { return }
+  async findAndRenderHouses(results, stat) {
+    if (stat == 'OK') {
+      if (results[0]) {
+        let lat = results[0].geometry.location.lat()
+        let lng = results[0].geometry.location.lng()
+        let houses = await HousesService.search({ lat, lng })
+        houses = this.housesFilters.filter(houses)
+        if (houses.data[0]) { 
+          this.placeMarkers(houses.data)
+          this.houses.render(houses.data)
+        } else {
+          alert('no results found')
+        }
+      } else {
+        alert('no results found')
+      }
+    } else { 
+      alert('no results found')
+    }
+  }
+
+  async placeMarkers(houses) {
+    let bounds = new google.maps.LatLngBounds()
     houses.forEach((house) => {
-      this.markers.push(
-        new google.maps.Marker({
-          position: {lat: house.latitude, lng: house.longitude},
-          map: this.map,
-        })
-      )
+      let marker = new google.maps.Marker({
+        position: {lat: house.latitude, lng: house.longitude},
+        map: this.map,
+        //icon: 'home.png'
+      })
+
+      this._setMarkerEventListener(marker, house)
+      bounds.extend(marker.position)
+      this.markers.push(marker)
     })
+    this.map.fitBounds(bounds)
+    this.map.setZoom(11)
   }
 
-  _resetMapChildren() {
+  _getInitialCoordinates() {
+    let initialCoords = new google.maps.LatLng(28.39404819, -91.38743867)
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        initialCoords = new google.maps.LatLng(position.coords.latitude, position.coords.longitude)
+        this.initMap(initialCoords)
+      })
+    }
+    return initialCoords
+  }
+
+  _removeMap() {
     while (this.mapParent.firstChild) {
       this.mapParent.removeChild(this.mapParent.firstChild)
     }
   }
 
+  _setMarkerEventListener(marker, house) {
+    let infoWindow = this._setInfoWindow(house)
```

Comment:
this stuff doesn't work when the google map is behind the dropdowns

Reply:
if it's in front of the dropdowns, mousing over the cursor displays an infowindow and clicking on it zooms in and displays it